lib/chapters_manager.php

```diff
@@ -22,106 +20,132 @@ public function __construct( Model\Episode $episode ) {
-	 * 
+	 *
 	 * @param  string $format object, psc, mp4chaps, json. Default: object
 	 * @return mixed
 	 */
 	public function get( $format = 'object' ) {
-
-		if ( ! $this->chapters_object )
-			$this->chapters_object = $this->get_chapters_object();
-
-		if ( ! $this->chapters_object )
-			return '';
+		$chapters_object = $this->get_chapters_object();
+		$chapters = '';
 
 		switch ( $format ) {
 			case 'psc':
-				$this->chapters_object->setPrinter( new Printer\PSC() );
-				return (string) $this->chapters_object;
+				$chapters_object->setPrinter( new Printer\PSC() );
+				$chapters = (string) $chapters_object;
 				break;
+
 			case 'mp4chaps':
-				$this->chapters_object->setPrinter( new Printer\Mp4chaps() );
-				return (string) $this->chapters_object;
+				$chapters_object->setPrinter( new Printer\Mp4chaps() );
+				$chapters = (string) $chapters_object;
 				break;
+
 			case 'json':
-				$this->chapters_object->setPrinter( new Printer\JSON() );
-				return (string) $this->chapters_object;
+				$chapters_object->setPrinter( new Printer\JSON() );
+				$chapters = (string) $chapters_object;
+				break;
+
+			case 'object':
+				$chapters = $chapters_object;
 				break;
 		}
 
-		return $this->chapters_object;
+		return apply_filters( 'podlove_get_chapters', $chapters, $format, $chapters_object );
 	}
 
-	private function get_raw_chapters_string() {
-		$asset_assignment = Model\AssetAssignment::get_instance();
+	protected function get_raw_chapters_string($chapters_file) {
 		$cache_key = 'podlove_chapters_string_' . $this->episode->id;
-		if ( ( $chapters_string = get_transient( $cache_key ) ) !== FALSE ) {
-			return $chapters_string;
-		} else {
-			if ( $asset_assignment->chapters == 'manual' ) {
-				return $this->episode->chapters;
-			} else {
-				if ( ! $chapters_asset = Model\EpisodeAsset::find_one_by_id( $asset_assignment->chapters ) )
-					return '';
-
-				if ( ! $chapters_file = Model\MediaFile::find_by_episode_id_and_episode_asset_id( $this->episode->id, $chapters_asset->id ) )
-					return '';
 
-				$chapters_string = wp_remote_get( $chapters_file->get_file_url() );
+		if ( false === ( $chapters_string = get_transient( $cache_key ) ) ) {
+			$request = wp_remote_get( $chapters_file->get_file_url() );
 
-				if ( is_wp_error( $chapters_string ) )
-					return '';
+			if ( is_wp_error( $request ) )
+				return '';
 
-				set_transient( $cache_key, $chapters_string['body'], 60*60*24*365 ); // 1 year, we devalidate manually
-				return $chapters_string['body'];
-			}
-		}	
+			// 1 year, we devalidate manually
+			$chapters_string = $request['body'];
+			set_transient( $cache_key, $chapters_string, 60*60*24*365 );
+		}
 
+		return $chapters_string;
 	}
 
 	private function get_chapters_object() {
+		global $wpdb;
+
+		// Get all posible assets with a chapter file format
+		$chapter_assets = Model\EpisodeAsset::find_all_by_sql(
+			"SELECT ea.*, ft.mime_type
+			 FROM {$wpdb->prefix}podlove_episodeasset AS ea
```

Comment:
Table names can be generated using `<Model Class>::table_name()`, for example `Model\EpisodeAsset::table_name()`. There should generally no need to directly access `$wpdb` when using the Publisher model classes.